### DISCUSSION TITLE
satellites

### DISCUSSION THREAD

**Discussion by nickrout** [JarvyJ/HomeIntent#249] [Support Questions] [answered]
Ho there. Love this project and look forward to installing. 

Can you give an idea how far away satellite support is likely to be? My home assistant is not in a living area, making a directly connected mic/speaker impractical except for testing.

And speaking of testing, I have a number of rpis and some respeaker gear which I can test with if that is helpful.

Meri kirihemete.

**Comment by JarvyJ** [comment]
I'm hoping to get satellites support in the next month or so! I was really shooting for December, but didn't have nearly as much time as I thought I did. I still need to do a bunch of testing with the [satellite images](https://github.com/JarvyJ/Rhasspy-Satellite) to ensure they work as expected and then a lot of frontend code to make it easier for folks to connect up satellites. Along with that, I really wanted to get a few other tasks in that'll make things easier down the line:

 * Refactoring the UI (#247)
 * Cleaning up the auto-restart behavior (#238)

So yeah, stay tuned, we'll work on getting it in in the next month or two! A bit earlier if you want to get into the nitty-gritty of [setting up Rhasspy as a satellite](https://rhasspy.readthedocs.io/en/latest/tutorials/#shared-mqtt-broker). I would also love to hear some feedback as to how it works with various pis and re-speakers!

**Reply by nickrout** [reply]
Thanks, I'll look into all those links.

**Reply by thundergreen** [reply]
I am using satellites already with Rhasspy and it seems working great so far. I have 3 M5 Atom Echo configured in Rhasspy and testing actually only with one. Once I got the impression everything is ready  to go,  I will add step by step the other two ones. But Satellites Works pretty good for the moment.

**Reply by JarvyJ** [reply]
@thundergreen, so satellites _can_ work with Home Intent (as you've found!), but certain things are broken (like timer responses #258). Also, right now, you need an externally managed Rhasspy to get it to work and I'm planning to get it working with a fully manged instance. 

Also, I've never heard of the m5 echo. I'll take a look, it seems like a cool little device

**Reply by thundergreen** [reply]
the m5 Echo is just awesome. You can flash custom code on it. Here the repo for flashing the echo to use as satellite:

https://github.com/Romkabouter/ESP32-Rhasspy-Satellite

It works surprisingly good. If there is any chance to make it a home assistant addon with all stuff in one .. well would be awesome :)

**Comment by JarvyJ** [comment, marked as answer]
As of the May release, we now have some [basic](https://homeintent.io/getting-started/advanced-features/satellites/) satellite support! There's also currently a PR (#392) to do some auto-configuration of satellites, so only downloading and installing Rhasspy on a pi would be needed (and Home Intent could handle the rest of it). It'll probably be available in the next couple of months.

**Reply by nickrout** [reply]
I just bought a couple of espmuse devices the other day and might try to give one a try as a satellite.
